native/dgesvd.go

```diff
@@ -0,0 +1,1328 @@
+							work[itauq:], work[itaup:], work[iwork:], lwork-iwork)
+
+						// Generate left bidiagonalizing vectors in work[ir:].
+						impl.Dorgbr(lapack.ApplyQ, n, n, n, work[ir:], ldworkr,
+							work[itauq:], work[iwork:], lwork-iwork)
+						iwork = ie + n
+
+						// Perform bidiagonal QR iteration, computing left singular
+						// vectors of R in work[ir:].
+						ok = impl.Dbdsqr(blas.Upper, n, 0, n, 0, s, work[ie:], dum, 1,
+							work[ir:], ldworkr, dum, 1, work[iwork:])
+
+						// Multiply Q in U by left singular vectors of R in
+						// work[ir:], storing result in A.
+						bi.Dgemm(blas.NoTrans, blas.NoTrans, m, n, n, 1, u, ldu,
+							work[ir:], ldworkr, 0, a, lda)
+
+						// Copy left singular vectors of A from A to U.
+						impl.Dlacpy(blas.All, m, n, a, lda, u, ldu)
+					} else {
+						// Insufficient workspace for a fast algorithm.
+						itau := 0
+						iwork := itau + n
+
+						// Compute A = Q*R, copying result to U.
+						impl.Dgeqrf(m, n, a, lda, work[itau:], work[iwork:], lwork-iwork)
+						impl.Dlacpy(blas.Lower, m, n, a, lda, u, ldu)
+
+						// Generate Q in U.
+						impl.Dorgqr(m, m, n, u, ldu, work[itau:], work[iwork:], lwork-iwork)
+						ie := itau
+						itauq := ie + n
+						itaup := itauq + n
+						iwork = itaup + n
+
+						// Zero out below R in A.
+						impl.Dlaset(blas.Lower, n-1, n-1, 0, 0, a[lda:], lda)
+
+						// Bidiagonalize R in A.
+						impl.Dgebrd(n, n, a, lda, s, work[ie:],
+							work[itauq:], work[itaup:], work[iwork:], lwork-iwork)
+
+						// Multiply Q in U by left bidiagonalizing vectors in A.
+						impl.Dormbr(lapack.ApplyQ, blas.Right, blas.NoTrans, m, n, n,
+							a, lda, work[itauq:], u, ldu, work[iwork:], lwork-iwork)
+						iwork = ie + n
+
+						// Perform bidiagonal QR iteration, computing left
+						// singular vectors of A in U.
+						ok = impl.Dbdsqr(blas.Upper, n, 0, m, 0, s, work[ie:],
+							dum, 1, u, ldu, dum, 1, work[iwork:])
+					}
+				} else if wantvo {
+					// Path 8.
+					panic(noSVDO)
+				} else if wantvas {
+					// Path 9.
+					if lwork >= n*n+max(max(n+m, 4*n), bdspac) {
+						// Sufficient workspace for a fast algorithm.
+						iu := 0
+						var ldworku int
+						if lwork >= wrkbl+lda*n {
+							ldworku = lda
+						} else {
+							ldworku = n
+						}
+						itau := iu + ldworku*n
+						iwork := itau + n
+
+						// Compute A = Q * R, copying result to U.
+						impl.Dgeqrf(m, n, a, lda, work[itau:], work[iwork:], lwork-iwork)
+						impl.Dlacpy(blas.Lower, m, n, a, lda, u, ldu)
+
+						// Generate Q in U.
+						impl.Dorgqr(m, m, n, u, ldu, work[itau:], work[iwork:], lwork-iwork)
+
+						// Copy R from A to VT, zeroing out below it.
+						impl.Dlacpy(blas.Upper, n, n, a, lda, vt, ldvt)
+						impl.Dlaset(blas.Lower, n-1, n-1, 0, 0, vt[ldvt:], ldvt)
+
+						ie = itau
+						itauq := ie + n
+						itaup := itauq + n
+						iwork = itaup + n
+
+						// Bidiagonalize R in VT.
```

Comment:
Perhaps it has been asked before but why is the rest of this if branch different from fortran?